gradle/java.gradle

```diff
@@ -69,20 +69,40 @@ repositories {
   mavenLocal()
 }
 
-dependencies {
-  compileOnly "org.checkerframework:checker-qual:${versions.checkerFramework}"
-
-  testImplementation enforcedPlatform("org.junit:junit-bom:${versions.junit5}")
-  testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api'
-  testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-params'
-  testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine'
-  testRuntimeOnly group: 'org.junit.vintage', name: 'junit-vintage-engine'
+evaluationDependsOn(":dependencyManagement")
+configurations {
+  dependencyManagement {
+    canBeConsumed = false
+    canBeResolved = false
+    visible = false
+  }
+}
+afterEvaluate {
+  configurations.configureEach {
+    if (canBeResolved && !canBeConsumed) {
+      extendsFrom(configurations.dependencyManagement)
+    }
+  }
+}
 
-  testImplementation deps.spock
-  testImplementation deps.groovy
-  testImplementation deps.testLogging
-  testImplementation "info.solidsoft.spock:spock-global-unroll:0.5.1"
-  testImplementation "com.github.stefanbirkner:system-rules:1.19.0"
+dependencies {
+  dependencyManagement platform(project(":dependencyManagement"))
+
+  compileOnly "org.checkerframework:checker-qual"
+
+  testImplementation "org.junit.jupiter:junit-jupiter-api"
+  testImplementation "org.junit.jupiter:junit-jupiter-params"
+  testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine"
+  testRuntimeOnly "org.junit.vintage:junit-vintage-engine"
+
+  testImplementation "org.objenesis:objenesis"
+  testImplementation "org.spockframework:spock-core"
+  testImplementation "ch.qos.logback:logback-classic"
```

Comment:
I didn't bother with deduping a lot of repetition related to logging, and also other libraries. Will follow up